set.mm

```diff
@@ -699314,31 +699320,66 @@ number axioms (add ~ ax-10 , ~ ax-11 , ~ ax-13 , ~ ax-nul , and remove
+      fveq2 oveq2 dvmptco ovif2 simpll abscld simplr absne0d reccld neg1cn 1cnd
+      mulcomd mulm1d divneg2d simpr ltled absnidd eqcomd negcon1ad oveq2d eqtrd
+      0red 3eqtrd sylanb recn ad2antrr rereccld mulridd cle simpl lenltd absidd
+      biimpar ifeqda eqtrid mpteq2ia eqtri ) EABAUAZUBUCZFUCZGHIZABJUULKIZUUKLU
+      DUEZJUFZJUGZMIZGZABJUUKKIZGUUNUUTNOADUULUURDUAZFUCZJUVBKIZEPUUMUUOUHUHBPQ
+      LUIIZUJZEEPUKZROULSPUVGROUMSOUUKBRZTZUOUVFUULUOLUNUPIZUVEUJZUVFUOUVJUVKUQ
+      UVJUVEURUSNUVKUVJNUVEUVJUSUVEUVJURUSNOQLUNQUTRZOVASLUTRZOVGSUNUTROVBSVCVD
+      VEUVJUVEVFVHVIUVJPVJUVKUVFVJLUNVKUVJPUVEVLVHVMUVIUUKUVHUUKPRZOUVHUUKUVHUU
+      KERZUUKLVNZUVHUUKELVOZUJZRUVOUVPTZBUVRUUKCVPUUKELVQVSZVRVTWAUVHUVPOUVHUVO
+      UVPUVTWBWAWCWDUURUHRUVIUUPUUQJJWEWFWJSOUVBUVFRZTZUVBUWAUVBPROUVBPUVEWGWAU
+      WAUVBLVNZOUWAUVBUVERZWKUWCUVBPUVEWHUWDUVBLUVBLNUWDLUVERZUVLUVMQLUDUEUWEVA
+      VGWIQLWLWMUVBLUVEWNWOWPWQWAWRUWBJUVBKXBEABUULGHIABUURGNOABCWSSPDUVFUVCGZH
+      IZDUVFUVDGZNOPFUVFWTZHIUWGUWHUWIUWFPHUWIUWFNODPUVQUJZFXAZUVFFUWJUWKFXCUWJ
+      UWKFXDOXEUWJUWKFXFXGUVFUWJVJOUVFUVFXHZXISXJVDXKDUVFUWLXLXMSUVBUULFXNUVBUU
+      LJKXOXPVDABUUSUVAUVHUUSUUPUUOUUQMIZUUOJMIZUGUVAUUPUUOUUQJMXQUVHUUPUWMUWNU
+      VAUVHUVSUUPUWMUVANUVTUVSUUPTZUWMUUQUUOMIUUOUFZUVAUWOUUOUUQUWOUULUWOUULUWO
+      UUKUWOUUKUVOUVPUUPXRZVTZXSVTZUWOUUKUWRUVOUVPUUPXTYAZYBZUUQPRUWOYCSYEUWOUU
+      OUXAYFUWOUWPJUULUFZKIUVAUWOJUULUWOYDUWSUWTYGUWOUXBUUKJKUWOUUKUULUWRUWOUUL
+      UUKUFUWOUUKUWQUWOUUKLUWQUWOYOUVSUUPYHYIYJYKYLYMYNYPYQUVHUVSUUPWKZUWNUVANU
+      VTUVSUXCTZUWNUUOUVAUXDUUOUXDUUOUXDUULUVOUULERUVPUXCUVOUUKUUKYRZXSYSUXDUUK
+      UVOUVNUVPUXCUXEYSUVOUVPUXCXTYAYTVTUUAUXDUULUUKJKUXDUUKUVOUVPUXCXRUVSLUUKU
+      UBUEUXCUVSLUUKUVSYOUVOUVPUUCUUDUUFUUEYMYNYQUUGUUHUUIUUJ $.
+  $}
+
+  ${
+    readvcot.d $e |- D = { y e. RR | ( sin ` y ) =/= 0 } $.
+    $( The support of sin ( ~ df-supp ) restricted to the reals is an open set.
+       (Contributed by SN, 7-Oct-2025.) $)
+    resuppsinopn $p |- D e. ( topGen ` ran (,) ) $=
+      ( csin cr cres ccnv cc cc0 csn cdif cfv co wcel ax-resscn mp2an crab wtru
+      ccn a1i cima ccnfld ctopn crest cioo crn ctg wss ccncf sincn eqid cncfcn1
+      eleqtri unicntop cnrest cnn0opn cnima cv wa resincl recnd adantr eldifsnd
+      wne eldifsni adantl impbida rabbiia cmpt wceq wf sinf feqresmpt mptpreima
+      simpr mptru 3eqtr4i tgioo4 3eltr4i ) DEFZGHIJKZUAZUBUCLZEUDMZBUEUFUGLVTWD
+      WCSMNZWAWCNWBWDNDWCWCSMZNEHUHZWEDHHUIMWFUJWCWCUKULUMOEDWCWCHUNUOPUPWAVTWD
+      WCUQPAURZDLZIVDZAEQWIWANZAEQBWBWJWKAEWHENZWJWKWLWJUSWIHIWLWIHNWJWLWIWHUTV
+      AVBWLWJVOVCWKWJWLWIHIVEVFVGVHCAEWIWAVTVTAEWIVIVJRAHHEDHHDVKRVLTWGROTVMVPV
+      NVQVRVS $.
+
+    $d D x $.  $d x y $.  $d x z $.
+    $( Real antiderivative of cotangent.  (Contributed by SN, 7-Oct-2025.) $)
+    readvcot $p |- ( RR _D ( x e. D |-> ( log ` ( abs ` ( sin ` x ) ) ) ) )
+                         = ( x e. D |-> ( ( cos ` x ) / ( sin ` x ) ) ) $=
```

Comment:
Did you see @david-a-wheeler has a definition of `cot` in his mathbox?

Reply:
Yes
I originally put down that `D` was `dom cot` and the RHS was `cot`, but then realized that the functions are restricted to the reals.